### PR TITLE
test: use go vet shadow feature instead of go-nyet

### PR DESCRIPTION
**Pull request by yichengq**
Use official support instead of home-made one.

/cc @barakmich

### CODE REVIEW

**Comment by xiang90** [comment]
LGTM.

File: test [hunk 1]
```diff
@@ -74,13 +74,18 @@ if [ -n "${vetRes}" ]; then
...
-if command -v go-nyet >/dev/null 2>&1; then
-  echo "Checking go-nyet..."
-  nyetRes=$(go-nyet -exitWith 0 $FMT)
-  if [ -n "${nyetRes}" ]; then
-          echo -e "go-nyet checking failed:\n${nyetRes}"
-          exit 255
-  fi
-fi
+echo "Checking govet -shadow..."
+for path in $FMT; do
```
Comment (timothysc) [inline test]: The FMT variable had an erroneous *.go addition above which can be removed iirc.

**Comment by timothysc** [comment]
minor comment otherwise lgtm.

File: test (hunk 1 above)
Comment (yichengq) [inline test]: It is helpful to check the format of go files in main directory IMO.

`go tool vet` doesn't support to check directories and files in one command line. If they are put in different command lines, that is fine.

Comment (timothysc) [inline test]: sgtm

File: raft/multinode.go [hunk 2]
```diff
@@ -283,16 +283,16 @@ func (mn *multiNode) run() {
 
 		case advs := <-advancec:
 			for groupID, rd := range advs {
-				group, ok := groups[groupID]
+				g, ok := groups[groupID]
```
Comment (yichengq) [inline raft/multinode.go]: /cc @bdarnell

**Pull request merged**